package-toolkit/runtime/src/main/kotlin/com/atlan/pkg/cache/DataDomainCache.kt

```diff
@@ -27,6 +29,37 @@ class DataDomainCache(val ctx: PackageContext<*>) : AssetCache<DataDomain>(ctx,
         if (result != null) cache(result.guid, getIdentityForAsset(result), result)
     }
 
+    private fun lookupByName(
+        name: String?,
+        currentAttempt: Int,
+        maxRetries: Int,
+    ): DataDomain? {
+        try {
+            val dataDomain =
+                DataDomain.select(client)
+                    .where(DataDomain.NAME.eq(name))
```

Comment:
This will only work for top-level domains (like `Marketing`). Domains can be nested, so a given domain's name could be something like `A@B@C@D@..`

You'd need to recursively lookup in the cache (and if a miss, retrieve by name) every one of those levels; and the same name at any given level could be elsewhere in a nested hierarchy, too:

The name `C` could appear at:

- `A@B@C`
- `X@Y@C`

^^ And even though at the lowest level the name is `C`, those are two completely different domains.